recipes/poco/all/conanfile.py

```diff
@@ -198,11 +200,13 @@ def package_info(self):
         if self.settings.os == "Linux":
             self.cpp_info.system_libs.extend(["pthread", "dl", "rt"])
 
-        if self.settings.compiler == "Visual Studio":
+        if self.settings.os == "Windows":
             self.cpp_info.defines.append("POCO_NO_AUTOMATIC_LIBS")
         if not self.options.shared:
             self.cpp_info.defines.append("POCO_STATIC=ON")
-            if self.settings.compiler == "Visual Studio":
+            if self.settings.os == "Windows":
                 self.cpp_info.system_libs.extend(["ws2_32", "iphlpapi", "crypt32"])
+                if self.options.enable_data_odbc:
+                    self.cpp_info.system_libs.extend(["odbc32", "odbccp32"])
```

Comment:
I think you should apply this to the correct component:
```python
if self.options.enable_data_odbc:
    if self.settings.os == "Windows":
        self.cpp_info.components["PocoDataODBC"].system_libs = ["odbc", "odbccp32"]
    else:
        self.cpp_info.components["PocoDataODBC"].requires = ["odbc::odbc"]
```

Reply:
In addition to to the "global" `cpp_info.system_libs` or replacing it?

Reply:
Replacing, because the other poco libraries should not require the odbc libraries.
You cannot use a global system_libs because poco uses components.
I modified it a bit to only link to the system libraries when creating a static library.

```python
if self.options.enable_data_odbc:
    if self.settings.os == "Windows":
        self.cpp_info.components["PocoDataODBC"].system_libs = ["odbc", "odbccp32"]
    else:
        if not self.options.shared:
            self.cpp_info.components["PocoDataODBC"].requires = ["odbc::odbc"]
```

Reply:
I'm lost, I don't see an components in package_info.

Reply:
Oh sorry!
Use global system_libs.
I mixed the dependency information with poco components..

Reply:
This is fine, you can mark this conversation as resolved.